src/parsers/transactions.ts

```diff
@@ -98,42 +103,41 @@ export const parseTransaction = (
 
   const fee = getTransactionFee(txn, nativeCurrency, chainId);
 
-  const contract = meta.contract_name && {
-    name: meta.contract_name,
-    iconUrl: meta.contract_icon_url,
+  const contract = meta?.contractName && {
+    name: meta.contractName,
+    iconUrl: meta.contractIconUrl,
   };
 
   // NOTE: For send transactions, the to address should be pulled from the outgoing change directly, not the txn.address_to
-  let to = txn.address_to;
-  if (meta.type === 'send') {
-    to = txn.changes.find(change => change?.direction === 'out')?.address_to ?? txn.address_to;
+  let to = txn.addressTo;
+  if (meta?.type === 'send') {
+    to = txn.changes.find(change => change?.direction === 'out')?.addressTo ?? txn.addressTo;
   }
 
   return {
     chainId,
-    from: txn.address_from,
+    from: txn.addressFrom,
     to,
-    title: buildTransactionTitle(type, status),
+    title: buildTransactionTitle(type, status as TransactionStatus),
     description,
     hash,
     network: txn.network,
     status,
     nonce,
-    protocol,
     type,
     direction,
     value,
     changes,
     asset,
-    approvalAmount: meta.quantity,
-    minedAt: txn.mined_at,
-    blockNumber: txn.block_number,
-    confirmations: txn.block_confirmations,
+    approvalAmount: meta?.quantity,
+    minedAt: minedAtTimestamp,
+    blockNumber: txn.blockNumber,
+    confirmations: txn.blockConfirmations,
     contract,
     native,
     fee,
-    explorerUrl: meta.explorer_url,
-    explorerLabel: meta.explorer_label,
+    explorerUrl: meta?.explorerUrl,
+    explorerLabel: meta?.explorerLabel,
   } as RainbowTransaction;
```

Comment:
nit: Nothing you changed, but should probably change this to `satisfies RainbowTransaction`